### ISSUE TITLE
Regression: Management of grade contributions lost

### ISSUE THREAD

**Issue by lordyavin** [theCrag/website#3538]
**What happened?**
I merged two routes into https://www.thecrag.com/climbing/south-africa/waterval-boven/route/18489865 and wanted to correct/check the grade contributions. Unfortunately it looks like the UI is gone to open the corresponding forms. 

**What you expected:**
Actually I can't remember the exact terms used in the UI but
* [ ] menu items to check/manage grade contributions

**Comment by scd** [comment]
This was removed intentionally because we believed they were no longer neccessary. Can you please explain in more detail why you need to check them in this way?

**Comment by lordyavin** [comment]
![grafik](https://user-images.githubusercontent.com/1528540/66386085-df8c8900-e9c1-11e9-8e47-c2fc2c1a6938.png)

![grafik](https://user-images.githubusercontent.com/1528540/66386102-e9ae8780-e9c1-11e9-92f5-ea2fa6da74e2.png)

![grafik](https://user-images.githubusercontent.com/1528540/66386174-06e35600-e9c2-11e9-9ac6-e82fa668db0b.png)

The Grade contribution of the 6b+ is not listed but displayed as overall grade.

**Comment by matozoid** [comment]
I've had several cases of merging routes, then finding they have a double grading where one of them is not accessible.

**Comment by scd** [comment]
You can actually update this with the standard interface. Unfortunately it is a bit of a pain because you have to change the grade to trigger the update. Each of the examples that have been supplied I had to change the grade to something else first then change it back.

**Comment by scd** [comment]
I am closing this.

**Comment by lordyavin** [comment]
The issue is not fixed for me. The grade contributed with the merged climb 6b+ is missing now. Is it lost?

**Comment by scd** [comment]
If that grade contribution is not there then it is more serious of a merging issue. 

@lordyavin do you want to rename this issue as a merge issue with grading where the assigned grades are different grading systems. The system should pick one then assign the other as a contribution. Whilst it is possible to have two assigned grades of different grading systems a lot of the code assumes this is not the case.

Yes I would say the 6b+ is lost.

**Comment by lordyavin** [comment]
Actually I miss the contribution management UI AND it looks like we need to improve the merging of routes. I wonder if the removed UI would have shown the merged grade (21 6b+).

I appreciate the approach to remove ambiguous UI where it is possible to do things in more than one way. But I don't  like it if we loose administrative features that help to manage the data.

**Comment by scd** [comment]
It looks like there is an edge case with the merging grades that is not taken into account. In particular where the two climbs have an assigned grade of the two routes has different grading systems one of the assigned grades is lost. 

It works if the assigned grade is of the same grading system.

It works for grade contributions.

The problem is I don't know what to do about this because a route should not have two assigned grades of different grading systems and there is no contributor to an assigned grade.

I have created an explicit issue for the failure case because there is too much noise here.

https://github.com/theCrag/website/issues/3827

Closing
